### ISSUE TITLE
[Feature Request]: Using SFSymbols as source for nativeImage

### ISSUE THREAD

**Issue by pie6k** [electron/electron#35711]
### Preflight Checklist

- [X] I have read the [Contributing Guidelines](https://github.com/electron/electron/blob/main/CONTRIBUTING.md) for this project.
- [X] I agree to follow the [Code of Conduct](https://github.com/electron/electron/blob/main/CODE_OF_CONDUCT.md) that this project adheres to.
- [X] I have searched the [issue tracker](https://www.github.com/electron/electron/issues) for a feature request that matches the one I want to file, without success.

### Problem Description

SFSymbols are macOS native icons consisting of 3k+ high quality icons. Currently seems there is no way to use them in Electron apps built for macOS.

<img width="691" alt="CleanShot 2022-09-16 at 21 47 10@2x" src="https://user-images.githubusercontent.com/7311462/190718766-6c6e1fe0-3407-44f5-b1f4-13ca7645f8c8.png">

It is trivial to use it natively in swiftUI

```swift
Image(systemName: "square.and.pencil")
```

### Proposed Solution

eg.

```ts
nativeImage.fromSFSymbolName("square.and.pencil")
```

### Alternatives Considered

It is possible to export SFSymbols to PNG using native MacOS app "SF Symbols", but it is quite a manual process.

**Comment by MarshallOfSound** [comment]
This is possible on macOS 11.0 and higher.  Important to note that your app if implemented your app will have to smooth fallbacks for these symbols anyway for a long time

https://developer.apple.com/documentation/appkit/nsimage/3622472-init

**Comment by ckerr** [comment]
In addition, SFSymbol licensing doesn't allow them to be used on non-Apple platforms, so it wouldn't make sense for Electron to do anything like bundling. Even if you use SFSymbols, you won't be able to make cross-platform Electron apps with them.

**Comment by pie6k** [comment]
I agree with your points about this not being cross-platform and working only on MacOS 11 or higher. 

However, I believe in many cases developers are aware and ok with those constraints. 

As SF Symbols are currently quite 'core' of 'native-feel-experience' on macOS platforms, I believe this could be added to Electron. There are already similar API's like https://www.electronjs.org/docs/latest/api/touch-bar which have very similar constraints.

Such an API `nativeImage.fromSFSymbolName` would be marked with `MacOS 11+` and it is up to developer to decide if this should be used if/how to provide fallback.

Note: inside 'BrowserWindow' - it is already relatively easy to use them as `SFSymbols` work a bit like emoji does if you use `SF Pro` font-family.

**Comment by szalaybalazs** [comment]
Electron already has a similar api for nativeImage: [createFromNamedImage](https://www.electronjs.org/docs/latest/api/native-image#nativeimagecreatefromnamedimageimagename-hslshift-macos)

In my opinion it would be worthwhile to extend this API to support sf symbols as well.

It is already marked as macOS only and it wouldn't need the introduction of a new API - only the extension of the current implementation.
